packages/kamino-sdk/src/Kamino.ts

```diff
@@ -847,22 +1012,16 @@ export class Kamino {
       reward1Vault: strategyState.reward1Vault,
       reward2Vault: strategyState.reward2Vault,
       poolRewardVault0:
-        strategyState.reward0Decimals.toNumber() > 0
-          ? whirlpool.rewardInfos[0].vault
-          : strategyState.baseVaultAuthority,
+        strategyState.reward0Decimals.toNumber() > 0 ? poolRewardVault0 : strategyState.baseVaultAuthority,
```

Comment:
Consider using toString instead of toNumber as well, just in case

Reply:
I replaced all the occurences of `toNumber`, I only kept it for decimals and `strategyType`